cmd/shortener/api/api.go

```diff
@@ -0,0 +1,79 @@
+package api
+
+import (
+	"fmt"
+	"github.com/KorsakovPV/shortener/cmd/shortener/config"
+	"github.com/go-chi/chi/v5"
+	"github.com/google/uuid"
+	"io"
+	"log"
+	"net/http"
+)
+
+type abstractStorage interface {
+	putURL(string) string
+	getURL(string) string
+}
+
+type localStorage struct{}
+
+func (s *localStorage) putURL(body string) string {
+	id := uuid.New().String()
+	shortURL[id] = body
+	return id
+}
+
+func (s *localStorage) getURL(id string) string {
+	return shortURL[id]
+}
+
+var (
+	shortURL = map[string]string{
+		"094c4130-9674-4c18-bf60-7385d7f61934": "https://practicum.yandex.ru/",
+	}
+)
+
+func createShortURL(rw http.ResponseWriter, r *http.Request) {
+
+	log.Println("Create short url")
+
+	bodyBytes, err := io.ReadAll(r.Body)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	var ls abstractStorage = &localStorage{}
+	id := ls.putURL(string(bodyBytes))
+
+	rw.Header().Set("Content-Type", "text/plain")
+	rw.WriteHeader(http.StatusCreated)
+
+	_, err = fmt.Fprintf(rw, "%s/%s", config.Config.FlagBaseURLAddr, id)
```

Comment:
P.S. https://stackoverflow.com/questions/37863374/whats-the-difference-between-responsewriter-write-and-io-writestring